### DISCUSSION TITLE
New Templates for ACE Card Views

### DISCUSSION THREAD

**Discussion by Annie-Johnson** [SharePoint/sp-dev-docs#8659] [Ideas]
### What type of issue is this?

Question

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Target SharePoint environment

SharePoint Online

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [X] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- Chrome Version 108.0.5359.99 (Official Build) (64-bit)
- SPFx version v1.13
- Node.js version * 14.18.1 (Currently using 64-bit executable)
- etc

### Issue description

A recent change to adaptive cards has rolled out to at least two of my tenants which removes the ability to override the template getter to customize a card view.  We were doing this for cards rendered in the SharePoint experience and rendering a different OOTB card for Teams mobile.  Clients are unhappy with OOTB design limitations.  Is there a safe, advisable way to continue supplying a third party card view?  I see the note on BaseCardView that third party card views should not inherit from that base class.  Is there a best practice way to develop a third party card view?

**Comment by ghost** [comment]
Thank you for reporting this issue. We will be triaging your incoming issue as soon as possible.

**Comment by johnguy0** [comment]
Unfortunately there is not a supported way to override the templates returned by the provided card views. The `template` methods on `BaseBasicCardView`, `BaseImageCardView`, and `BasePrimaryTextCardView` are marked as `@sealed` to designate they should not (cannot) be overridden. Additionally there is no supported way to create additional card views beyond those provided. 

I understand these may be limiting. However, we want to enforce a coherent design language on the dashboard. We are actively working on expanding the set of out-of-box card views in the future. If there are particular requirements or designs you'd like to see, please feel free to suggest them!

**Comment by AJIXuMuK** [comment]
Converting this issue to the discussion as the behavior is by design but we want to learn more about custom templates

**Comment by Annie-Johnson** [comment]
Thanks John and Alex.  Happy to share some insight into the card features we'd like to have available.  First, a color picker for the text and optional background image would go a long way for our clients.  It would also be nice to have ID attributes on the cards and card elements, so that clients could choose to customize what is available at their own risk to fit in better with the workspaces we are designing with web parts and application customizers.  

A welcome card that accepts an Azure maps API URL to show weather data, allowing a link through to the user's profile.

![image](https://user-images.githubusercontent.com/28032142/208179088-d08aad6f-4d34-4832-afab-9ed26f8684b1.png)

Events cards are popular with our clients, along with the background image with bottom aligned larger text card.  You can also see how being able to hook into IDs helps us make small aesthetic tweaks such as rounding the border radius and changing button background colors.  

![image](https://user-images.githubusercontent.com/28032142/208187944-e010fa87-bb57-46d5-98dc-86d317a73fa4.png)

![image](https://user-images.githubusercontent.com/28032142/208186429-da092798-3a65-46ca-a87f-ca8fa627f9cb.png)

Taken on the whole within the workspace this allows for a dashboard that is more consistent with the greater experience design and is more aligned to a client's brand identity.  We understand that we need to deliver a paired down, OOTB version of the card in the Teams mobile experience.  Clients are accepting of that, and it would be nice to support rendering a different card view on mobile vs desktop out of the box as well.  

Another fairly in demand layout is this 'my links' card..  support for up to 6 audience targeted links, so that we can get a launching point for the users' most important tools and resources in front of them without needing to take up the space of 6 individual cards

![image](https://user-images.githubusercontent.com/28032142/208190933-279d3e23-22f5-437e-8e9b-e8613556af6b.png)


Finally, clients would really love to have the option to hide a card dynamically.  An example use case would be a card that appears to congratulate an employee on their work anniversary or to commemorate a special event that feeds off a back-end list.  On days when there is no special event to display, the card would not appear on the dashboard.

I have many other custom card use cases that I could share, but I think the more important message is that there is a big demand for increased design flexibility.  I have clients who have invested $$$$ in custom brand-driven workspaces that are put off by the Viva Dashboard because it can't be more flexible.  I would encourage you to consider allowing us to extend the card view as a backlog item. 

Please let me know if you'd like to continue this conversation, I'm happy to share additional insight into the demands we hear from clients and how they are using their dashboards today.

**Reply by johnguy0** [reply]
Thanks for the detailed write up @Annie-Johnson! I don't have any additional specifics to share just yet, but I will say we have been taking a lot of feedback (this included) into our planning for the next couple SPFx releases. Feedback like yours in invaluable to helping us define the best capabilities we can provide through SPFx. 

I don't think we're at the point yet of opening up full card customization. We do understand the current template set needs to evolve and grow. 

I would love to hear more about how customers are using the Viva Connections dashboard and what scenarios we can unblock from the platform.

**Comment by BassemNKhalil** [comment]
Would it be possible to bring 'up' the adaptive card content into the ACE itself i.e. the ACE renders with this [Flight Itinerary](https://adaptivecards.io/samples/FlightItinerary.html) sample instead of an ACE with a button that shows the flight itinerary on click?

**Reply by johnguy0** [reply]
This unfortunately isn't something we have planned currently, but please keep bubbling it up as feedback. We're trying to juggle expressiveness with cohesive design practices on our dashboards. If there's particular designs you'd like to see made available through our templates, please provide that feedback.

**Comment by ir3ne** [comment]
Hey John!
I'll look at it again tuesday sisnce I'm in vacation. So more feedbacks
next and if you want to see my professional skills see:
irenetomaini.com

Best!

Irene

Il giorno ven 20 gen 2023 alle 23:19 John Nguyen ***@***.***>
ha scritto:
